src/main/kotlin/org/jetbrains/intellij/tasks/RunIdePerformanceTestTask.kt

```diff
@@ -0,0 +1,122 @@
+package org.jetbrains.intellij.tasks
+
+import com.jetbrains.plugin.structure.base.utils.createDir
+import com.jetbrains.plugin.structure.base.utils.exists
+import org.gradle.api.Incubating
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.property
+import org.jetbrains.intellij.IntelliJPluginExtension
+import org.jetbrains.intellij.getIdeJvmArgs
+import org.jetbrains.intellij.info
+import org.jetbrains.intellij.error
+import org.jetbrains.intellij.logCategory
+import org.jetbrains.intellij.model.ProfilerName
+import org.jetbrains.intellij.performanceTest.parsers.IdeaLogParser
+import org.jetbrains.intellij.performanceTest.model.PerfTestResult
+import org.jetbrains.intellij.performanceTest.parsers.SimpleIJPerfParser
+import org.jetbrains.intellij.performanceTest.TestExecutionFailException
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+@Incubating
+open class RunIdePerformanceTestTask : RunIdeTask() {
+
+    private val context = logCategory()
+
+    /**
+     * Path to directory with test projects and '.ijperf' files.
+     */
+    @get:Input
+    val testDataDir = objectFactory.property<String>()
+
+    /**
+     * Path to artifacts directory(idea logs, snapshots, screenshots etc.) will be placed. If directory isn't exist, it will be created automatically.
```

Comment:
Path to directory where performance test artifacts (IDE logs, snapshots, screenshots, etc.) will be stored.
If directory doesn't exist, it will be created